src/festim/problem.py

```diff
@@ -120,8 +120,16 @@ def create_solver(self):
             bcs=self.bc_forms,
         )
         self.solver = NewtonSolver(MPI.COMM_WORLD, problem)
-        self.solver.atol = self.settings.atol
-        self.solver.rtol = self.settings.rtol
+        self.solver.atol = (
+            self.settings.atol
+            if not callable(self.settings.rtol)
+            else self.settings.rtol(0.0)
+        )
+        self.solver.rtol = (
+            self.settings.rtol
+            if not callable(self.settings.rtol)
+            else self.settings.rtol(0.0)
+        )
```

Comment:
At this point in the simulation, if `self.t` is defined, could we use `float(self.t)` instead of hard coding 0.0?